server/Arcadia.Assistant.CSP/Vacations/EmployeeVacationApprovedAccountingReminderActor.cs

```diff
@@ -0,0 +1,207 @@
+                this.Self,
+                RemindVacations.Instance,
+                this.Self);
+
+            Context.System.EventStream.Subscribe<CalendarEventChanged>(this.Self);
+            Context.System.EventStream.Subscribe<CalendarEventRecoverComplete>(this.Self);
+        }
+
+        public static Props CreateProps(string employeeId, AccountingReminderConfiguration reminderConfiguration)
+        {
+            return Props.Create(() => new EmployeeVacationApprovedAccountingReminderActor(
+                employeeId,
+                reminderConfiguration));
+        }
+
+        protected override void OnReceive(object message)
+        {
+            switch (message)
+            {
+                case CalendarEventChanged msg when
+                    msg.NewEvent.EmployeeId == this.employeeId &&
+                    msg.NewEvent.Type == CalendarEventTypes.Vacation:
+
+                    this.OnEventReceived(msg.NewEvent);
+                    break;
+
+                case CalendarEventChanged _:
+                    break;
+
+                case CalendarEventRecoverComplete msg when
+                    msg.Event.EmployeeId == this.employeeId &&
+                    msg.Event.Type == CalendarEventTypes.Vacation:
+
+                    this.OnEventReceived(msg.Event);
+                    break;
+
+                case CalendarEventRecoverComplete _:
+                    break;
+
+                case RemindVacations _:
+                    this.SendReminderNotifications()
+                        .PipeTo(
+                            this.Self,
+                            success: result => new RemindVacations.Success(result),
+                            failure: err => new RemindVacations.Error(err));
+                    break;
+
+                case RemindVacations.Success msg:
+                    if (msg.PushNotifications.Any())
+                    {
+                        this.logger.Debug($"Sending push reminder notifications about approved vacations of employee {this.employeeId}");
+                    }
+
+                    foreach (var pushNotification in msg.PushNotifications)
+                    {
+                        Context.System.EventStream.Publish(new NotificationEventBusMessage(pushNotification));
+                    }
+
+                    break;
+
+                case RemindVacations.Error msg:
+                    this.logger.Error(
+                        "Error occured on send push reminder notifications about approved vacations " +
+                        $"of employee {this.employeeId}: {msg.Exception.Message}");
+                    break;
+
+                default:
+                    this.Unhandled(message);
+                    break;
+            }
+        }
+
+        private void OnEventReceived(CalendarEvent @event)
+        {
+            if (@event.Status != VacationStatuses.Approved)
+            {
+                if (this.remindedEvents.ContainsKey(@event.EventId))
```

Comment:
not remindedEvents in this case but vacationsToRemind?

Reply:
Done.